### DISCUSSION TITLE
Firmware Update & Connection Issues

### DISCUSSION THREAD

**Discussion by jprinsen** [addiejanssen/domoticz-solaredge-modbustcp-plugin#30] [Q&A]
SolarEdge has pushed new firmware to my SE3000H Inverter causing my Modbus TCP data connection to freeze since yesterday afternoon. As a result, domoticz data has not been updated since. The current version numbers are:
DSP2: 2.100.1017
DSP1: 1.100.1146
CPU: 4.14.228

The logfile (screendump attached) states:
```
- Pushing 'onHeartbeatCallback' on to the queue
- Retrying to communicate with inverter after: [date/time]
- Processing 'onHeartbeatCallback' message
- Acquiring lock for 'onHeartbeatCallback'
- Calling message handler 'onHeartbeat' on 'module' type object
- onHeartbeat
- Retrying to comunicate with inverter after: [date/time]
- Releasing to lock for 'onHeartbeatCallback'; 
- followed by a repetitive string or acquiring/releasing lock for 'SolarEdge'...
```
![CleanShot 2022-05-18 at 09 47 53@2x](https://user-images.githubusercontent.com/24796548/168987762-ad02d925-95f6-4e65-be7d-385cbec79503.png)

I verified the system settings; ModBusTCP is still enabled; also the port number has not changed.
Still running the same (and most recent) pymodbus version (2.5.3) so we can exclude that item as well.

Anybody with some useful insights in what is causing this and how to re-establish my valuable SolarEdge data connection?
Many thanks in anticipation.

**Comment by jprinsen** [comment]
It appears that CPU version 4.14.XXX no longer allows ModBus TCP connections over WiFi. Is not mentioned in the changelog  since 'ModbusTCP via WiFi' was never formally considered to be a 'feature' (hear-say). Will look into possibilities to enable ModBusTCP again via LAN; since my SE3000 only has WiFi I would welcome any suggestion by any of you.

**Reply by jovo71** [reply]
I think i face the same issue. 
It would be great to learn how to solve this.
Pulling a wired connection isnt really an option.

**Reply by addiejanssen** [reply]
Could you try updating the plugin?
I have increased the connection timeout value; not sure if that fixes the issue, but worth trying.
